### PR TITLE
Version v0.0.7

### PR DESCRIPTION
**Pull request by PhMueller**
This Pull Request combines the currently open PRs. (Dont need to merge them anymore).

* **ConfigFile**: Improve version checking
* **XGBoost**: Add benchmark with parameter `booster`
* **Cartpole**: Reduce number of unecessary warnings
* **Registry:** Host container on Gitlab
* **AbstractContainerBenchmark**: Interface takes `container_tag` as parameter
* **AbstractBenchmark**: Remove inactive parameter from the configuration
* **README**: Fix fidelity name 

Note: It is the update to version 0.0.7 (not 0.0.8)

### CODE REVIEW

File: hpobench/abstract_benchmark.py [hunk 1]
```diff
@@ -131,17 +132,37 @@ def _check_and_cast_configuration(configuration: Union[Dict, ConfigSpace.Configu
...
         else:
             raise TypeError(f'Configuration has to be from type List, np.ndarray, dict, or '
                             f'ConfigSpace.Configuration but was {type(configuration)}')
+
+        all_hps = set(configuration_space.get_hyperparameter_names())
+        active_hps = configuration_space.get_active_hyperparameters(configuration)
+        inactive_hps = all_hps - active_hps
+
+        if len(inactive_hps) != 0:
+            logger.debug(f'There are inactive {len(inactive_hps)} hyperparameter: {inactive_hps}'
```
Comment (KEggensperger) [inline hpobench/abstract_benchmark.py:159]: Could you add a test for this?

File: hpobench/abstract_benchmark.py [hunk 2]
```diff
@@ -131,17 +132,37 @@ def _check_and_cast_configuration(configuration: Union[Dict, ConfigSpace.Configu
                                       configuration_space: ConfigSpace.ConfigurationSpace) \
             -> ConfigSpace.Configuration:
         """ Helper-function to evaluate the given configuration.
-            Cast it to a ConfigSpace.Configuration and evaluate if it violates some boundaries.
+            Cast it to a ConfigSpace.Configuration and evaluate if it violates its boundaries.
+
+            Note:
+                We remove inactive parameters from the configuration. Inactive parameters are parameters that are not
```
Comment (KEggensperger) [inline hpobench/abstract_benchmark.py]: We remove inactive hyperparameters from the given configuration. Inactive hyperparameters are hyperparameters that are not relevant for a configuration, e.g. A is only relevant if B=1 and if B!=1 then A is inactive will be removed from the configuration if present.

File: hpobench/benchmarks/surrogates/paramnet_benchmark.py [hunk 3]
```diff
@@ -44,6 +44,26 @@
 pip install .[paramnet]
 
 ```
+
+Changelog:
+==========
+0.0.2:
```
Comment (KEggensperger) [inline hpobench/benchmarks/surrogates/paramnet_benchmark.py:50]: Why is the changelog duplicated? L48-L56 and L58-L66

File: hpobench/benchmarks/surrogates/paramnet_benchmark.py [hunk 4]
```diff
@@ -269,17 +291,22 @@ def objective_function(self, configuration: Union[CS.Configuration, Dict],
 
         return {'function_value': float(y),
                 'cost': fidelity['budget'],
-                'info': {'learning_curve': lc.tolist(),
-                         'observed_epochs': len(lc)}}
+                'info': {'fidelity': fidelity['budget'],
```
Comment (KEggensperger) [inline hpobench/benchmarks/surrogates/paramnet_benchmark.py]: L276 and L294 should be the same? I assume both should be `'info': {'fidelity': fidelity,`

File: hpobench/container/recipes/ml/Singularity.PyBNN [hunk 5]
```diff
@@ -11,7 +11,7 @@ VERSION v0.0.1
     && cd /home \
     && git clone https://github.com/automl/HPOBench.git \
     && cd HPOBench \
-    && git checkout development \
+    && git checkout Fix_PyBnn \
```
Comment (KEggensperger) [inline hpobench/container/recipes/ml/Singularity.PyBNN]: This is not correct, right?

File: hpobench/container/recipes/surrogates/Singularity.ParamnetBenchmark [hunk 6]
```diff
@@ -12,6 +12,7 @@ VERSION v0.0.1
     cd /home \
     && git clone https://github.com/automl/HPOBench.git \
     && cd HPOBench \
+    && git checkout Fix_ParamNetOnTime \
```
Comment (KEggensperger) [inline hpobench/container/recipes/surrogates/Singularity.ParamnetBenchmark]: same here

**Review by KEggensperger — CHANGES_REQUESTED** [review]
Left some comments

File: hpobench/benchmarks/surrogates/paramnet_benchmark.py (hunk 4 above)
Comment (PhMueller) [reply hpobench/benchmarks/surrogates/paramnet_benchmark.py]: Thanks! Should I also add the entries: 

```
'learning_curve': []
'observed_epochs': 0
```
to the info dict of L274?

Comment (KEggensperger) [reply hpobench/benchmarks/surrogates/paramnet_benchmark.py]: Good point. Yes, please.

**Pull request merged**